src/iarc7_motion/iarc_tasks/hit_roomba_task.py

```diff
@@ -0,0 +1,219 @@
+                    return (TaskDone(),)
+
+                try:
+                    roomba_transform = self._tf_buffer.lookup_transform(
+                                        'level_quad',
+                                        self._roomba_id,
+                                        rospy.Time.now(),
+                                        rospy.Duration(self._TRANSFORM_TIMEOUT))
+                except (tf2_ros.LookupException,
+                        tf2_ros.ConnectivityException,
+                        tf2_ros.ExtrapolationException) as ex:
+                    rospy.logerr('ObjectTrackTask: Exception when looking up transform')
+                    rospy.logerr(ex.message)
+                    return (TaskAborted(msg='Exception when looking up transform during hit roomba'),)
+
+                # Creat point centered at drone's center
+                stamped_point = PointStamped()
+                stamped_point.point.x = 0
+                stamped_point.point.y = 0
+                stamped_point.point.z = 0
+
+                # returns point distances of roomba to center point of level quad
+                self._roomba_point = tf2_geometry_msgs.do_transform_point(
+                                                    stamped_point, roomba_transform)
+
+                # uses _roomba_point to determine if quad too far from roomba
+                if not self._check_max_roomba_range():
+                    return (TaskAborted(msg='The provided roomba is not close enough to the quad'),)
+                
+                roomba_x_velocity = self._roomba_odometry.twist.twist.linear.x
+                roomba_y_velocity = self._roomba_odometry.twist.twist.linear.y
+
+                # p-controller
+                x_vel_target = (self._roomba_point.point.x * self._K_X + roomba_x_velocity)
+                y_vel_target = (self._roomba_point.point.y * self._K_Y + roomba_y_velocity)
+                
+                z_vel_target = self._descent_velocity
+
+                #caps velocity
+                vel_target = math.sqrt(x_vel_target**2 + y_vel_target**2)
+
+                if vel_target > self._MAX_HORIZ_SPEED:
+                    x_vel_target = x_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                    y_vel_target = y_vel_target * (self._MAX_HORIZ_SPEED/vel_target)
+                
+                if (abs(z_vel_target) > self._MAX_Z_VELOCITY):
+                    z_vel_target = z_vel_target/abs(z_vel_target) * self._MAX_Z_VELOCITY
+
+                desired_vel = [x_vel_target, y_vel_target, z_vel_target]
+
+                drone_vel_x = self._drone_odometry.twist.twist.linear.x
+                drone_vel_y = self._drone_odometry.twist.twist.linear.y
+                drone_vel_z = self._drone_odometry.twist.twist.linear.z
+
+                curr_vel = [drone_vel_x, drone_vel_y, drone_vel_z]
+
+                desired_vel = self._limiter.limit_acceleration(curr_vel, desired_vel)
```

Comment:
This should use the current velocity only the first time because if we use the current velocity every time the velocity controller might take a long time to accelerate at all since you won't necessarily get the acceleration you request. Especially small ones.